libensemble/generators.py

```diff
@@ -104,7 +117,7 @@ def ask(self, num_points: Optional[int] = 0) -> List[dict]:
 
     def tell(self, results: List[dict]) -> None:
         """Send the results of evaluations to the generator."""
-        self.tell_numpy(list_dicts_to_np(results))
+        self.tell_numpy(list_dicts_to_np(results))  # OH, we need the union of sim_specs.out and gen_specs.out
```

Comment:
If using externally may not even have sim_specs, or gen_specs.

Reply:
Exactly. I think we briefly discussed on Tuesday that we have to do the conversion anyway; and thankfully libE can handle receiving numpy arrays with dtypes it previously was unaware of

Reply:
For the external interface, I would think each gen must need to say name/type for dictionaries entries (in VOCS). So user knows what they must supply. But they only allow scalars and something like sampling can take varying dimensions, what would the interface be. x0, x1, x2 ... Xn